otherlibs/stdune-unstable/env.ml

```diff
@@ -77,9 +77,17 @@ let add t ~var ~value = make (Map.set t.vars var value)
 
 let remove t ~var = make (Map.remove t.vars var)
 
-let extend t ~vars = make (Map.superpose t.vars vars)
-
-let extend_env x y = extend x ~vars:y.vars
+let extend t ~vars =
+  if Map.is_empty vars then
+    t
+  else
+    make (Map.superpose t.vars vars)
```

Comment:
I wonder if it's really worth doing this micro-optimisation. `Map.superimpose` doesn't do much work in the `Map.is_empty` case, so is this mostly for saving an allocation in `make`?

Reply:
This and also keeping the cached `unix` value.

Reply:
This and also keeping the cached `unix` value.

Reply:
But ideally, we could tweak the API in spawn so that we don't need this, i.e. take a map directly.

Reply:
OK, let's leave it as is.